hoot-services/src/main/java/hoot/services/utils/DbUtils.java

```diff
@@ -829,6 +828,85 @@ public static long getRecordIdForInputString(String input, RelationalPathBase<?>
         return -1;
     }
 
+    public static String getJobIdByTask(String taskInfo) {
+        String jobId = createQuery()
+                .select(jobStatus.jobId)
+                .from(jobStatus)
+                .where(Expressions.booleanTemplate("tags->'taskInfo' like '%" + taskInfo + "%'"))
+                .fetchFirst();
+
+        return jobId;
+    }
+
+    public static List<Long> getTimeoutTasks() {
+        List<String> list = createQuery()
+            .select(Expressions.stringTemplate("tags->'taskInfo'"))
+            .from(jobStatus)
+            .where(Expressions.booleanTemplate("exist(tags,'timeout')"))
+            .fetch();
+
+        String patternString = "taskingManager:([0-9]*)_([0-9]*)";
+        Pattern pattern = Pattern.compile(patternString);
+        List<Long> taskList = new ArrayList<>();
+        for (String taskTag: list) {
+            Matcher matcher = pattern.matcher(taskTag);
+            boolean found = matcher.find();
+            long taskId = -1;
+            if (found) {
+                taskId = Long.parseLong(matcher.group(2));
+            }
+
+            taskList.add(taskId);
+        }
+
+        return taskList;
+    }
+
+    public static Map<String, String> getJobStatusTags(String jobId) {
+        Object jobTags = createQuery().select(jobStatus.tags)
+                .from(jobStatus)
+                .where(jobStatus.jobId.eq(jobId))
+                .fetchOne();
+
+        return PostgresUtils.postgresObjToHStore(jobTags);
+    }
+
+    public static void tagTimeoutTask(String jobId) {
+        Map<String, String> tags = getJobStatusTags(jobId);
+        tags.put("timeout", "true");
+
+        SQLQueryFactory query = createQuery();
+        query.update(jobStatus)
+            .where(jobStatus.jobId.eq(jobId))
+            .set(jobStatus.tags, tags)
+            .execute();
+
+        try {
+            query.getConnection().commit();
+        }
+        catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
```

Comment:
This commit code is necesary because down the line here we do another update and I think this cause some weird hanging issue: https://github.com/ngageoint/hootenanny/blob/master/hoot-services/src/main/java/hoot/services/job/JobStatusManagerImpl.java#L359

I have a feeling this is also related to the problem I saw a while ago with command status and how I had to change how I wrote the sql for that to prevent the same issue